src/transactionWatcher.ts

```diff
@@ -47,89 +61,101 @@ export class TransactionWatcher {
 
-        return this.awaitConditionally<ITransactionOnNetwork>(
-            foundAllEvents,
-            doFetch,
-            errorProvider
-        );
+        return this.awaitConditionally<ITransactionOnNetwork>(foundAllEvents, doFetch, errorProvider);
     }
 
-    public async awaitAnyEvent(txHash: string, events: string[]): Promise<ITransactionOnNetwork> {
+    public async awaitAnyEvent(
+        transactionOrTxHash: ITransaction | string,
+        events: string[],
+    ): Promise<ITransactionOnNetwork> {
         const foundAnyEvent = (transactionOnNetwork: ITransactionOnNetwork) => {
-            const allEventIdentifiers = this.getAllTransactionEvents(transactionOnNetwork).map(event => event.identifier);
-            const anyIsFound = events.find(event => allEventIdentifiers.includes(event)) != undefined;
+            const allEventIdentifiers = this.getAllTransactionEvents(transactionOnNetwork).map(
+                (event) => event.identifier,
+            );
+            const anyIsFound = events.find((event) => allEventIdentifiers.includes(event)) != undefined;
             return anyIsFound;
         };
 
-        const doFetch = async () => await this.fetcher.getTransaction(txHash);
+        const doFetch = async () => {
+            const hash = this.transactionOrTxHashToTxHash(transactionOrTxHash);
+            return await this.fetcher.getTransaction(hash);
+        };
         const errorProvider = () => new ErrExpectedTransactionEventsNotFound();
 
-        return this.awaitConditionally<ITransactionOnNetwork>(
-            foundAnyEvent,
-            doFetch,
-            errorProvider
-        );
+        return this.awaitConditionally<ITransactionOnNetwork>(foundAnyEvent, doFetch, errorProvider);
     }
 
-    public async awaitOnCondition(txHash: string, condition: (data: ITransactionOnNetwork) => boolean): Promise<ITransactionOnNetwork> {
-        const doFetch = async () => await this.fetcher.getTransaction(txHash);
+    public async awaitOnCondition(
+        transactionOrTxHash: ITransaction | string,
+        condition: (data: ITransactionOnNetwork) => boolean,
+    ): Promise<ITransactionOnNetwork> {
+        const doFetch = async () => {
+            const hash = this.transactionOrTxHashToTxHash(transactionOrTxHash);
+            return await this.fetcher.getTransaction(hash);
+        };
         const errorProvider = () => new ErrExpectedTransactionStatusNotReached();
 
-        return this.awaitConditionally<ITransactionOnNetwork>(
-            condition,
-            doFetch,
-            errorProvider
-        );
+        return this.awaitConditionally<ITransactionOnNetwork>(condition, doFetch, errorProvider);
+    }
+
+    private transactionOrTxHashToTxHash(transactionOrTxHash: ITransaction | string): string {
+        return typeof transactionOrTxHash === "string" ? transactionOrTxHash : transactionOrTxHash.getHash().hex();
```

Comment:
could also check the validity of the hash. it can be "invalid tx hash" instead of 64 hex chars

Reply:
Done